jmix-fullcalendar/fullcalendar-flowui/src/main/java/io/jmix/fullcalendarflowui/component/data/LazyCalendarEventProvider.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.fullcalendarflowui.component.data;
+
+import io.jmix.fullcalendarflowui.component.FullCalendar;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.TimeZone;
+
+/**
+ * Interface to be implemented by event providers that should load items by request.
+ * <p>
+ * The {@link #onItemsFetch(ItemsFetchContext)} is called in the following cases:
+ * <ul>
+ *     <li>
+ *         When the user navigates to the calendar view that has wider date range than previous view.
+ *     </li>
+ *     <li>
+ *         When {@link FullCalendar}'s navigation methods are invoked, e.g. {@link FullCalendar#navigateToNext()}.
+ *     </li>
+ *     <li>
+ *          When time zone of component is changed.
+ *     </li>
+ * </ul>
+ * By default {@link FullCalendar} calls {@link #onItemsFetch(ItemsFetchContext)} immediately after adding
+ * the provider. If calendar is not attached to the UI, loading will be performed after attaching component
+ * to the UI.
+ *
+ * @see LazyEntityCalendarEventRetriever
+ * @see CalendarEventRetriever
+ */
+public interface LazyCalendarEventProvider extends BaseCalendarEventProvider {
```

Comment:
Let's name it `CallbackCalendarEventProvider`, similar to `com.vaadin.flow.data.provider.CallbackDataProvider` that is used in `HasLazyDataView`

Reply:
I named it CallbackCalendar**Data**Provider to maintain consistency with other providers.